packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/controllers/EditingContextTextBasedObjectsDataFetcher.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.editingcontext.controllers;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.sirius.components.annotations.spring.graphql.QueryDataFetcher;
+import org.eclipse.sirius.components.collaborative.dto.EditingContextTextBasedObjectsPayload;
+import org.eclipse.sirius.components.collaborative.dto.EditingContextTextBasedObjectsInput;
+import org.eclipse.sirius.components.graphql.api.IDataFetcherWithFieldCoordinates;
+import org.eclipse.sirius.components.graphql.api.IEditingContextDispatcher;
+
+import graphql.schema.DataFetchingEnvironment;
+import reactor.core.publisher.Mono;
+
+/**
+ * Data fetcher for the field EditingContext#textBasedObjects.
+ *
+ * @author gdaniel
+ */
+@QueryDataFetcher(type = "EditingContext", field = "textBasedObjects")
```

Comment:
Why is this code here and not in `org.eclipse.sirius.components.core.graphql.datafetchers`?

Reply:
Done